### PR TITLE
refactor: app logic in `router`

### LINKED ISSUE #3187: [Task]: Refactor `router` module
### Task description

Currently there is a lot of logic contained within the application routers, keeping strictly routing logic.

### Requirements

- [x] #3221
- [x] Remove profile migration out of `login-router.ts` (15min)
- [x] Move app logic out of `backup-router.ts` (15min)
- [x] Move app logic out of `protect-router.ts` (15min)
- [x] Move app logic out of `app-router.ts` (30min)

### Acceptance criteria

Meets the above requirements.

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary
The code within the router needs to be refactored and updated according to the onboarding flow changes.

### Changelog
```
- Removes application logic from the router module (so that it contains routing logic ONLY)
- Cleanup some code in route file for onboarding
```

## Relevant Issues
Closes #3187 

## Type of Change
- [ ] __Breaking__ - any change that would cause existing functionality to not work as expected
- [x] __Chore__ - refactoring, build scripts or anything else that isn't user-facing
- [ ] __Docs__ - changes to the documentation
- [ ] __Fix__ - a change which fixes an issue
- [ ] __New__ - a change which implements a new feature
- [x] __Update__ - a change which updates existing functionality

### Platforms
- __Desktop__
	- [ ] MacOS
	- [x] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

### Instructions
Navigate to all corners of the app and see that the router is working as intended.

## Checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or modified tests that prove my changes work as intended
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by Tuditi — CHANGES_REQUESTED** [review]
I'm encountering some errors upon import, I'll have to look into it a bit more

File: packages/shared/lib/core/profile/actions/active-profile/saveActiveProfile.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export function saveActiveProfile(): void {
             id: _activeProfile?.id,
             name: _activeProfile?.name,
             type: _activeProfile?.type,
+            version: _activeProfile?.version,
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply packages/shared/lib/core/profile/actions/active-profile/saveActiveProfile.ts:13]: Did the problems occur on this branch?

File: packages/shared/routes/setup/backup/views/BackupToFile.svelte [hunk 2]
```diff
@@ -1,32 +1,47 @@
...
     let confirmPassword
-    let skipping = false
+    let skipBackup = false
 
     const dispatch = createEventDispatcher()
 
-    $: valid = strongholdPassword === confirmPassword
+    $: isStrongholdPasswordValid = strongholdPassword === confirmPassword
+
+    function handleBackClick() {
```
Comment (Tuditi) [inline packages/shared/routes/setup/backup/views/BackupToFile.svelte:21]: Missing return type

File: packages/shared/routes/setup/backup/views/BackupToFile.svelte [hunk 3]
```diff
@@ -1,32 +1,47 @@
...
 
-    function onSubmit() {
-        if (valid) {
-            skipping = false
+        await storeMnemonic($mnemonic.join(' '))
+        if (!_skipBackup) {
+            await backupInitialStronghold()
             dispatch('next')
+        } else {
+            dispatch('next', { skip: true })
```
Comment (Tuditi) [inline packages/shared/routes/setup/backup/views/BackupToFile.svelte:33]: Maybe switch the if, else statement to remove the negation?

**Review by Tuditi — APPROVED** [review]
I added some minor comments, but I'll approve so that you're not blocked to merge it. Nice work! :man_dancing:

File: packages/shared/lib/core/profile/actions/active-profile/saveActiveProfile.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tuditi) [reply packages/shared/lib/core/profile/actions/active-profile/saveActiveProfile.ts:13]: Yes, they also occur on stardust-develop

File: packages/shared/routes/setup/backup/views/BackupToFile.svelte (hunk 3 above)
Comment (maxwellmattryan) [reply packages/shared/routes/setup/backup/views/BackupToFile.svelte:33]: Yeah :sweat_smile:

**Pull request merged**